torch/optim/adamax.py

```diff
@@ -70,7 +77,7 @@ def _init_group(self, group, params_with_grad, grads, exp_avgs, exp_infs, state_
 
             # State initialization
             if len(state) == 0:
-                state["step"] = torch.tensor(0.0)
+                state["step"] = torch.tensor(0.0, device=p.device)
```

Comment:
This is likely going to cause perf regressions for eager as it now will incur kernel launches over python math.

Reply:
yep, I'm not sure how to fix the `_inductor.ir.ComputeBuffer.get_device() == "cpu"` any other way, though.

Reply:
As mentioned in https://github.com/pytorch/pytorch/pull/110339, adam also uses CPU tensors in eager mode but somehow these are traced away by the time they get to inductor :thinking:

Reply:
https://github.com/pytorch/pytorch/blob/428cbd7513267335d0c321ed1a438f9bbb1e150a/torch/_dynamo/variables/optimizer.py#L40

I see, it seems that dynamo will try to make all params capturable. Unsure why this is legal… I seem to recall that certain combinations (capturable+for each+differentiator) will assert fail…

Reply:
Used the recommended "capturable" flag strategy in https://github.com/pytorch/pytorch/pull/110339. Will add it here shortly.